chain/jsonrpc-primitives/src/blocks.rs

```diff
@@ -0,0 +1,43 @@
+#[derive(thiserror::Error, Debug)]
+pub enum RpcBlockError {
+    #[error("Block `{0}` is missing")]
+    BlockMissing(near_primitives::hash::CryptoHash),
+    #[error("Block not found")]
+    BlockNotFound(String),
+    #[error("There are no fully synchronized blocks yet")]
+    NotSyncedYet,
+    #[error("Too many requests. Try again later: {0}")]
+    TooManyRequests(actix::MailboxError),
+    #[error("Unexpected error occurred: {0}")]
+    Unexpected(String),
```

Comment:
I suggest the following renaming:

* `Unexpected` -> `Unreachable` - this is the common term in Rust, and we should indicate in the error message that we would love to be informed about the error if it happens (e.g. if it only happens on someone else's instance, where we don't collect metrics), so the error message could be something along these lines: "It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose"
* `TooManyRequests` -> `InternalError` with a help message: "The node reached its limits. Try again later. More details: {0}"